### PR TITLE
fix(types): Fix defaultConfig typing for makeQueryCache

### LINKED ISSUE #806: bug(types): `defaultConfig` for `makeQueryCache` is typed incorrectly
**Describe the bug**

I was wondering why the `defaultConfig` wasn't merging correctly with my query configs and it turns out it is just a typing issue.

**To Reproduce**

```
  const testQueryCache = makeQueryCache({
    defaultConfig: {
      queries: {
        refetchOnMount: false,
      },
    },
  });
```

**Expected behavior**

No TS error occurs

**Screenshots**

```
Type '{ queries: { refetchOnMount: false; }; }' is not assignable to type 'BaseQueryOptions<unknown, Error>'.
  Object literal may only specify known properties, and 'queries' does not exist in type 'BaseQueryOptions<unknown, Error>'.ts(2322)
```

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

### PR DESCRIPTION
**Pull request by kamranayub**
Fixes #806 

## Changes

- Fix `defaultConfig` typing of `makeQueryCache`

### CODE REVIEW

File: types/test.ts [hunk 1]
```diff
@@ -39,36 +42,39 @@ function prefetchQuery() {
 }
 
 function getQueryData() {
-  queryCache.getQueryData(['queryKey']);
-  queryCache.getQueryData('queryKey');
-  queryCache.getQueryData(true);
-  queryCache.getQueryData((query) => true);
+  queryCache.getQueryData(['queryKey'])
```
Comment (kamranayub) [inline types/test.ts:45]: Hmm, seems prettier took effect here

Comment (kamranayub) [reply types/test.ts:45]: @tannerlinsley would you prefer I revert these formatting changes?

**Comment by kamranayub** [comment]
Tests seem to be failing with unrelated issues, probably #751

**Comment by kamranayub** [comment]
See comment https://github.com/tannerlinsley/react-query/issues/806#issuecomment-666435081 but even with TS source, this will be an issue until the declaration files are generated since the package typings resolve to `types/index.d.ts` still.

**Comment by tannerlinsley** [comment]
I'll merge for now.

**Pull request merged**